include/dlaf/interface/cholesky.h

```diff
@@ -0,0 +1,105 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <mpi.h>
+#include <dlaf/interface/cholesky.h>
+
+#include <pika/init.hpp>
+#include <dlaf/factorization/cholesky.h>
+#include <dlaf/interface/blacs.h>
+#include <dlaf/interface/utils.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_mirror.h>
+
+namespace dlaf::interface {
+
+template <typename T>
+using MatrixMirror = dlaf::matrix::MatrixMirror<T, dlaf::Device::Default, dlaf::Device::CPU>;
+
+template <typename T>
+void pxpotrf(char uplo, T* a, int m, int n, int mb, int nb, int lld, const MPI_Comm& communicator,
+             int nprow, int npcol) {
```

Comment:
This should be an internal algorithm API, that is used by the C interface.
Anyway it is C++, so it can be declared and defined in `include/dlaf/cholesky.h`

Reply:
Moved to private headers in `src/c_api`, as suggested below.

> Any private interface (e.g. the templated pxpotrf) should be added either to the source files or to private headers in the src/c_api tree.